apps/deploy-web/src/components/billing-usage/PaymentMethodsContainer/PaymentMethodsContainer.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useCallback, useState } from "react";
+
+import { usePaymentMethodsQuery, usePaymentMutations, useSetupIntentMutation } from "@src/queries";
+import type { PaymentMethodsViewProps } from "../PaymentMethodsView/PaymentMethodsView";
+
+const DEPENDENCIES = {
+  usePaymentMethodsQuery,
+  usePaymentMutations,
+  useSetupIntentMutation
+};
+
+type PaymentMethodsContainerProps = {
+  children: (props: PaymentMethodsViewProps) => React.ReactNode;
+  dependencies?: typeof DEPENDENCIES;
+};
+
+export const PaymentMethodsContainer: React.FC<PaymentMethodsContainerProps> = ({ children, dependencies: d = DEPENDENCIES }) => {
+  const { data: paymentMethods = [], isLoading: isLoadingPaymentMethods, refetch: refetchPaymentMethods } = d.usePaymentMethodsQuery();
+  const { setPaymentMethodAsDefault, removePaymentMethod } = d.usePaymentMutations();
+  const { data: setupIntent, mutate: createSetupIntent, reset: resetSetupIntent } = d.useSetupIntentMutation();
+  const [showAddPaymentMethod, setShowAddPaymentMethod] = useState(false);
+
+  const onSetPaymentMethodAsDefault = useCallback((id: string) => {
```

Comment:
we have an agreement that function names should reflect an action not a trigger. So in this case it should be:
```suggestion
  const setPaymentMethodAsDefault = useCallback((id: string) => {
```

It would conflict with the const above so I see 2 approaches. 
- avoid destructuring methods from usePaymentMutations and use them as `paymentMutations.setPaymentMethodAsDefault`
- use `setPaymentMethodAsDefault.mutate` directly in the html callback prop, which in this case is also fine as it's just a single liner